src/ngx_http_lua_uthread.c

```diff
@@ -273,6 +273,15 @@ ngx_http_lua_uthread_kill(lua_State *L)
         ngx_http_lua_del_thread(r, L, ctx, sub_coctx);
         ctx->uthreads--;
 
+        /* notify the parent that the thread was killed */
+        parent_coctx = sub_coctx->parent_co_ctx;
+        if (sub_coctx->waited_by_parent && parent_coctx) {
```

Comment:
@aviramc I think there is a corner case that a child light thread kills its parent and the parent is waiting on this child. We need to clear the `sub_coctx->waited_by_parent` field here at least, right? Since nobody is waiting on this killer thread any more. Thoughts?